easybuild/main.py

```diff
@@ -543,6 +543,7 @@ def process_eb_args(eb_args, eb_go, cfg_settings, modtool, testing, init_session
     # build software, will exit when errors occurs (except when testing)
     if not testing or (testing and do_build):
         exit_on_failure = not (options.dump_test_report or options.upload_test_report)
+        run_hook(POISED, hooks, args=[ordered_ecs])
```

Comment:
Why not make this call *in* `build_and_install_software` instead?

Reply:
Looking at this, I believe it would require another argument containing the hooks to be passed to `build_and_install_software`. Could be wrong though.

Reply:
You can add `hooks = load_hooks(eb_go.options.hooks)` in `build_and_install_software`, which is fine since the loaded hooks are cached - see https://github.com/easybuilders/easybuild-framework/blob/b8c153df1a9299d256a482413b5bca0321e72e1b/easybuild/tools/hooks.py#L109

Reply:
On second thought, let's leave it as is, since the way it's done now makes the relation to `build_and_install_software` very clear, especially since there's now a `pre_build_and_install_loop` and `post_build_and_install_loop`